twitter/Views/TweetCell.m

```diff
@@ -7,18 +7,130 @@
 //
 
 #import "TweetCell.h"
+#import "APIManager.h"
+#import "DateTools.h"
+
+@interface TweetCell()
+
+@property (weak, nonatomic) IBOutlet UIImageView *profilePicture;
+@property (weak, nonatomic) IBOutlet UILabel *userDisplayName;
+@property (weak, nonatomic) IBOutlet UILabel *userAccountName;
+@property (weak, nonatomic) IBOutlet UILabel *displayDate;
+@property (weak, nonatomic) IBOutlet UITextView *tweetText;
+
+
+@property (weak, nonatomic) IBOutlet UIButton *retweetButton;
+@property (weak, nonatomic) IBOutlet UILabel *retweetCountLabel;
+@property (weak, nonatomic) IBOutlet UIButton *favoriteButton;
+@property (weak, nonatomic) IBOutlet UILabel *likeCountLabel;
+@property (weak, nonatomic) IBOutlet UIButton *replyButton;
+@property (weak, nonatomic) IBOutlet UILabel *replyCountLabel;
+
+@end
 
 @implementation TweetCell
 
 - (void)awakeFromNib {
     [super awakeFromNib];
-    // Initialization code
 }
 
 - (void)setSelected:(BOOL)selected animated:(BOOL)animated {
     [super setSelected:selected animated:animated];
+}
+
+- (void)refreshData:(Tweet *)tweet {
+    self.cellTweet = tweet;
+    [self refreshCurrentCell];
+}
+
+- (void)refreshCurrentCell {
```

Comment:
The reason this is a bad pattern -

Right now the mapping of fields on the tweet to 'fields' on the cell are pretty much 1:1. There is little data munging / transformations / formatting etc... what happens when the business logic changes? When the format of the incoming tweet changes? What happens when you need to pass something on the tweet through say, a privacy filter? Now you have a ton of business logic in your view where it doesn't belong.